File: 15.BinarySearch二分查找/leetcode_3876_uniformArray.py
from bisect import *
from sortedcontainers import SortedList

class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        cnt = SortedList()
        for i in nums1:
            if i&1:
                cnt.add(i)
        def calc(x):
            for i in nums1:
                if i%2 == x:
                    continue
                indx = cnt.bisect_left(i)-1
                if indx<0 or i-cnt[indx]<1:
                    return False
            return True
        return calc(0) or calc(1)

if __name__ == "__main__":
     s = Solution()
     nums = [1,4,7]
     print(s.uniformArray(nums))

File: 10.Math数学/leetcode_3876_uniformArray.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int minEven = INT_MAX, minOdd = INT_MAX;
        for (int i : nums1){
            if (i&1){
                minOdd = min(minOdd, i);
            }
            else{
                minEven = min(minEven, i);
            }
        }
        if (minOdd == INT_MAX || minEven == INT_MAX){
            return true;
        }
        if (minOdd < minEven){
            return true;
        }
        return false;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,4,7};
    cout << s.uniformArray(nums) << endl;
}
